### PR TITLE
Merge modify-list-command branch

### PR DESCRIPTION
**Pull request by carinackwu**

### CODE REVIEW

**Comment by carinackwu** [comment]
fix #83

File: src/main/java/seedu/todolist/logic/commands/ListCommand.java [hunk 1]
```diff
@@ -8,15 +11,38 @@
...
+    	if (dateFilter.isEmpty()) {
+    		model.updateFilteredListToShowAll();
+            return new CommandResult(MESSAGE_ALLTASKS_SUCCESS);
+    	}
+    	else if (!(dateFilter.equals("today") || dateFilter.equals("week") || dateFilter.equals("month"))
+    			&& !isValidDate(dateFilter)) {
+            return new CommandResult(MESSAGE_FILTER_INVALID);
+    	}
+    	
+		try {
```
Comment (AngShiYa) [inline src/main/java/seedu/todolist/logic/commands/ListCommand.java:37]: Indentation problem

File: src/main/java/seedu/todolist/logic/commands/ListCommand.java [hunk 2]
```diff
@@ -8,15 +11,38 @@
...
 
     @Override
     public CommandResult execute() {
-        model.updateFilteredListToShowAll();
-        return new CommandResult(MESSAGE_SUCCESS);
+    	if (dateFilter.isEmpty()) {
+    		model.updateFilteredListToShowAll();
+            return new CommandResult(MESSAGE_ALLTASKS_SUCCESS);
+    	}
+    	else if (!(dateFilter.equals("today") || dateFilter.equals("week") || dateFilter.equals("month"))
```
Comment (AngShiYa) [inline src/main/java/seedu/todolist/logic/commands/ListCommand.java:32]: Magic string. Use a constant.

File: src/main/java/seedu/todolist/model/ModelManager.java [hunk 3]
```diff
@@ -239,4 +248,68 @@ public String toString() {
...
+        public boolean run(ReadOnlyTask task) throws DateTimeException {
+        	if (task.getInterval().isFloat()) {
+        		return false;
+        	}
+        	
+        	LocalDate currentDate = LocalDate.now();
+        	
+        	int currentDayOfWeek = currentDate.getDayOfWeek().getValue();
+        	LocalDate currentWeekStart = currentDate.minusDays(currentDayOfWeek);
+        	LocalDate currentWeekEnd = currentDate.plusDays(6-currentDayOfWeek);
```
Comment (AngShiYa) [inline src/main/java/seedu/todolist/model/ModelManager.java:268]: Why 6? Use a constant

File: src/main/java/seedu/todolist/model/ModelManager.java [hunk 4]
```diff
@@ -239,4 +248,68 @@ public String toString() {
...
+                	return (!taskEndDate.isBefore(currentMonthStart) && !taskEndDate.isAfter(currentMonthEnd));
+                } 
+                else {
+                    LocalDate date = DateParser.parseDate(dateFilter);
+                    return taskEndDate.equals(date);
+                }
+        	}
+        	else {
+        		LocalDate taskStartDate = task.getInterval().getStartDate().getDate();
+            	LocalDate taskEndDate = task.getInterval().getEndDate().getDate();
```
Comment (AngShiYa) [inline src/main/java/seedu/todolist/model/ModelManager.java:292]: Indentation problem again :(

**Review by AngShiYa — COMMENTED** [review]
Looks good. Just need to be careful about the indentation and magic number/string.

**Pull request merged**